### DISCUSSION TITLE
How would I make "hostname replace" replace https to http?

### DISCUSSION THREAD

**Discussion by Gabryx64** [searxng/searxng#1294] [Q&A] [answered]
Is there a way to make searxng's hostname replace change the https to an http too? i do
```yaml
hostname_replace:
  '(.*\.)?youtube\.com$': 'localhost:7270'
  '(.*\.)?youtu\.be$': 'localhost:7270'
```
which does replace the hostname but not the https, that i need to be http since the website is hosted locally and it is not https.
I tried doing
```yaml
hostname_replace:
  'https://(.*\.)?youtube\.com$': 'http://localhost:7270'
  'https://(.*\.)?youtu\.be$': 'http://localhost:7270'
```
but it does not work probably because it only checks for the the hostname.

**Comment by return42** [comment, marked as answer]
Present we do not have an official solution.  The "hostname_replace" plugin isn't the Swiss-Army-Knife we need for efficiently filtering (weight, rewrite, etc) URLs in the result list and the info pages (and other elements). 

We also do not yet have a solution that would generate the corresponding CSP header.  At the moment, the setups with "hostname_replace" are manual work that is insufficiently integrated into SearXNG.

TL;DR;

"hostname_replace" is a very primitive plugin, we need a mature solution in SearXNG (not a plugin). --> I have tagged all posts on this topic by [URL rewrite & filter](https://github.com/search?q=repo%3Asearxng%2Fsearxng%20label%3A%22URL%20rewrite%20%26%20filter%22%20&type=issues)
